debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -0,0 +1,166 @@
+ */
+package io.debezium.transforms.outbox;
+
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.RegexRouter;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+/**
+ * Debezium Outbox Transform Event Router
+ *
+ * @author Renato mefi (gh@mefi.in)
+ */
+public class EventRouter<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventRouter.class);
+
+    private final ExtractField<R> afterExtractor = new ExtractField.Value<>();
+    private final RegexRouter<R> regexRouter = new RegexRouter<>();
+    private EventRouterConfigDefinition.InvalidOperationBehavior invalidOperationBehavior;
+
+    private String fieldEventId;
+    private String fieldEventKey;
+    private String fieldEventType;
+    private String fieldPayload;
+    private String fieldPayloadId;
+    private String fieldPayloadType;
+
+    private String routeByField;
+
+    @Override
+    public R apply(R r) {
+        // Ignoring tombstones
+        if (r.value() == null) {
+            LOGGER.info("Tombstone message {} ignored", r.key());
+            return null;
+        }
+
+        Struct debeziumEventValue = requireStruct(r.value(), "Detect Debezium Operation");
+        String op = debeziumEventValue.getString(Envelope.FieldName.OPERATION);
+
+        // Skipping deletes
+        if (op.equals(Envelope.Operation.DELETE.code())) {
+            LOGGER.info("Delete message {} ignored", r.key());
+            return null;
+        }
+
+        // Dealing with unexpected update operations
+        if (op.equals(Envelope.Operation.UPDATE.code())) {
+            handleUnexpectedOperation(r);
+            return null;
+        }
+
+        final R afterRecord = afterExtractor.apply(r);
+        Struct eventStruct = requireStruct(afterRecord.value(), "Read Outbox Event");
+
+        Long timestamp = debeziumEventValue.getInt64("ts_ms");
+
+        String eventId = eventStruct.getString(fieldEventId);
+        String eventType = eventStruct.getString(fieldEventType);
+        String payload = eventStruct.getString(fieldPayload);
+        String payloadId = eventStruct.getString(fieldPayloadId);
+        String payloadType = eventStruct.getString(fieldPayloadType);
+
+        Schema valueSchema = SchemaBuilder.struct()
```

Comment:
Can we cache the schema? I remember building it can be costly from earlier SMTs.